lib/authorizer/basic.js

```diff
@@ -43,7 +44,7 @@ module.exports = {
      * @todo - this should be made into ".sign" after the signature calculation is moved out of the SDK.
      */
     sign: function (params, request) {
-        if (!params || !params.username) { return request; } // Nothing to do if no parameters are present.
+        if (!params || !_.has(params, 'username') || !_.has(params, 'password')) { return request; }
```

Comment:
instead of checking for truthy value (since the empty string is not truthy, it was bailing out), we now only check for the existence of the property
Should we also check the type to be a string?

Reply:
Empty strings are still valid, property existence checks will suffice here.

Reply:
yeah, this PR makes empty string valid!
I was asking whether we should add a check that the username and password parameter MUST be a string (not an object or Array)

Reply:
Yes, do add the string like check.